tests/spec/formatter/text-format.spec.tsx

```diff
@@ -1,19 +1,25 @@
 import * as React from 'react';
-import * as moment from 'moment';
+import * as dayjs from 'dayjs';
 import { mount } from 'enzyme';
 import { expect } from 'chai';
 import * as numeral from 'numeral';
 require('numeral/locales'); // load numeral-js locale data
+require('dayjs/locale/it');
```

Comment:
what about other locales, can we dynamically load this based on the language selected?

Reply:
ah never mind, this is a test, but how does this get loaded for app?

Reply:
I was thinking that it is loaded by generator-jhipster but I can't find it.
So I'm wondering if it is missing and we should update generator-jhipster to add the `required()` required.
Or maybe react-jhipster should do it ? (but I think react-jhipster should just provide the component and not load the appropriate locales)
I will think about it...

Reply:
Languages are loaded here: https://github.com/jhipster/generator-jhipster/blob/main/generators/generator-base-private.js#L424

Reply:
so what's the state of this ticket ? should it be merged ?

Reply:
I will do several tests with a generated app, and play with locale and format to see if everything is still working well before eventually merging. Thx

Reply:
So good catch because even if in the main app we import the locale (ex: `import 'dayjs/locale/vi';`), this import is not "visible" in the `TextFormat` component. (maybe because it's a dependency ? and thus a different "scope", I don't know much..)
So I've added `require('dayjs/locale/' + locale);` at runtime during the rendering of the component.

Reply:
May be use dynamic import instead of require to keep it clean

Reply:
Does this mean to use `import('dayjs/locale/' + locale);` instead of `require('dayjs/locale/' + locale);` ?
It is working but the rendering of the translation is late.
Example: A date is first rendered as 'October' and a few moment later it is rendered as its correct translation 'tháng 10'.
This is because import() is asynchronous. I will keep investigating on this.